### ISSUE TITLE
[Unvalidated Sign-Up] Include a magic link in every unread message summary footer and use it for any CTAs in the email as well

### ISSUE THREAD

**Issue by anmurali** [Expensify/App#34166]
**Problem**:
Email notifications from New Dot have links to go to new.expensify.com to respond, or approve, or pay etc. 
Clicking on this simply brings you to new.expensify.com like any other site visitor and you have to sign in from scratch. 
This feels like a sub optimal experience since we already know the user (we emailed them), and we know where they need to go (we included the CTA in the email), and yet we force them to sign in from scratch and navigate to where they need to go on their own

**Solution**:
1. IF the account is unvalidated (ie, nobody has ever signed in)
- THEN make every link we send via UnreadMessageSummary into a magic link (because there's nothing in the account yet to steal)
2. However, after you click it the first time, you have an infinite session, so in theory you shouldn't need a magic link when clicking the link the next time.  This would mean:
- The first link you click drops you straight into the product
- The second link requires authentication with a magic link, if you don't already have an infinite session (eg, if you are on a new device)
3. Use that same logic for any link that would be used for any "calls to action" in the body of the email, which will soon be refactored to look exactly like the product UI.

**Comment by MitchExpensify** [comment]
Looking for help here https://expensify.slack.com/archives/CC7NECV4L/p1704845562604679?thread_ts=1704818282.056719&cid=CC7NECV4L

**Comment by techievivek** [comment]
I can work on this one. Thanks. 🙌

**Comment by MitchExpensify** [comment]
Asking for an update here https://expensify.slack.com/archives/C05NJ4SLBMF/p1705701594874099

**Comment by techievivek** [comment]
Sorry, I couldn't prioritize this today. I will work on it tomorrow :for-sure:, I also replied to the thread here: https://expensify.slack.com/archives/C05NJ4SLBMF/p1706595259842319

**Comment by techievivek** [comment]
Update for today: https://expensify.slack.com/archives/C05NJ4SLBMF/p1706667468085419

**Comment by techievivek** [comment]
Added an updated here: https://expensify.slack.com/archives/C05NJ4SLBMF/p1707141774434789?thread_ts=1706915414.620219&cid=C05NJ4SLBMF

**Comment by techievivek** [comment]
Not overdue, I created a GH for the frontend changes https://github.com/Expensify/App/issues/36325, and posted an updated here: https://expensify.slack.com/archives/C05NJ4SLBMF/p1707530916220149

**Comment by techievivek** [comment]
App PR is deployed to staging and Web-E PR is ready for review https://github.com/Expensify/Web-Expensify/pull/40734

**Comment by techievivek** [comment]
Web-E PR is merged 🎉, posted an updated in Slack here: https://expensify.slack.com/archives/C05NJ4SLBMF/p1708135487521139

**Comment by MitchExpensify** [comment]
Shall we close or what else are we waiting on?

**Comment by trjExpensify** [comment]
👋 I think we should track this issue along with the "native" sign-up not requiring validation/magic code, so I'm going to bring it under the `[Unvalidated Sign-up]` project and bump up the priority to the August release. I don't think we necessarily need this one for XeroCon, but we should minimally have it re-implemented for Q2 where we expect to see more invited users to NewDot. 

On that topic, @techievivek you might be interested [in this](https://expensify.slack.com/archives/C036QM0SLJK/p1715770663165699?thread_ts=1715336674.469839&cid=C036QM0SLJK). It would be super valuable for the XeroCon conference, so let us know there if you're interested in getting that work over the line. Thanks!

**Comment by trjExpensify** [comment]
I'm going to remove the `Reviewing` label on this issue, as a new PR is required to re-implement this one and I can't see it linked. I'd probably also suggest just closing https://github.com/Expensify/Expensify/issues/374479 and using this one where the context of what we're trying to achieve is in the OP from Anu. 👍

**Comment by techievivek** [comment]
As discussed here https://expensify.slack.com/archives/C036QM0SLJK/p1716211040969329?thread_ts=1715336674.469839&cid=C036QM0SLJK, I will review this and the main GH https://github.com/Expensify/App/issues/30794 along with the linked PR by tomorrow.

**Comment by quinthar** [comment]
Excited for this!  Let's chat in realtime here: https://expensify.slack.com/archives/C066HJM2CAZ/p1716892649705859

**Comment by techievivek** [comment]
Still discussing potential solution on above thread.

**Comment by techievivek** [comment]
Web-E PR here: https://github.com/Expensify/Web-Expensify/pull/42241

**Comment by techievivek** [comment]
Auth PR is deployed to PROD. Web-E we will get it merge it today.

**Comment by techievivek** [comment]
This is done.
